### PR TITLE
fix(FEC-13661): Move plugins to type script - kaltura/playkit-js-kava

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

solves FEC-13661

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,26 +1,60 @@
...
-  },
-  "globals": {
-    "should": true,
-    "sinon": true,
-    "__VERSION__": true,
-    "__NAME__": true,
-    "process": true,
-    "__dirname": true
-  },
+  "root": true,
```
Comment (SivanA-Kaltura) [inline .eslintrc.json:2]: why change line rules ?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
 ### Description of the Changes
 
-Please add a detailed description of the change, weather it's an enhancement or a bugfix.
+Please add a detailed description of the change, whether it's an enhancement or a bugfix.
 If the PR is related to an open issue please link to it.
 
-### CheckLists
+**Issue:**
```
Comment (SivanA-Kaltura) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: why change this file

File: .prettierrc [hunk 3]
```diff
@@ -1,12 +1,7 @@
...
   "tabWidth": 2,
-  "useTabs": false,
-  "semi": true,
   "singleQuote": true,
   "trailingComma": "none",
-  "bracketSpacing": false,
-  "jsxBracketSameLine": true,
-  "arrowParens": "avoid",
-  "proseWrap": "preserve"
+  "arrowParens": "always"
```
Comment (SivanA-Kaltura) [inline .prettierrc:6]: why change this file

File: karma.conf.js [hunk 4]
```diff
@@ -1,44 +1,86 @@
...
+
+    // Concurrency level
+    // how many browser should be started simultaneous
+    concurrency: Infinity,
+
     client: {
       mocha: {
         reporter: 'html',
-        timeout: 12000
+        timeout: 50000
```
Comment (SivanA-Kaltura) [inline karma.conf.js:82]: why ?

File: package.json [hunk 5]
```diff
@@ -3,97 +3,73 @@
...
-    "karma-ie-launcher": "^1.0.0",
+    "@babel/core": "^7.23.7",
+    "@babel/plugin-transform-runtime": "^7.23.7",
+    "@babel/preset-env": "^7.23.8",
+    "@babel/preset-typescript": "^7.23.3",
+    "@babel/runtime": "^7.23.8",
+    "@microsoft/api-extractor": "^7.39.1",
+    "@playkit-js/browserslist-config": "1.0.8",
+    "@playkit-js/kaltura-player-js": "^3.17.10",
+    "@playkit-js/moderation": "canary",
```
Comment (SivanA-Kaltura) [inline package.json:37]: use explicit versions

File: test/e2e/kava.spec.js [hunk 6]
```diff
@@ -1,10 +1,11 @@
-import '../../src/index.js';
+// import '../../src/index.js';
```
Comment (SivanA-Kaltura) [inline test/e2e/kava.spec.js:1]:
```suggestion
```

File: src/temp-imported-plugins-event-names-temp.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
+export const DownloadEvent = {
```
Comment (SivanA-Kaltura) [inline src/temp-imported-plugins-event-names-temp.ts:1]: what is this file for ?

File: src/kava.ts [hunk 8]
```diff
@@ -865,45 +863,48 @@ class Kava extends BasePlugin {
...
-  _initTabMode(): void {
+  private _initTabMode(): void {
     let hiddenAttr: string;
     let visibilityChangeEventName: string;
     if (typeof document.hidden !== 'undefined') {
       // Opera 12.10 and Firefox 18 and later support
       hiddenAttr = 'hidden';
       visibilityChangeEventName = 'visibilitychange';
+      // @ts-expect-error - Property 'msHidden' does not exist on type 'Document'. Did you mean 'hidden'?
     } else if (typeof document.msHidden !== 'undefined') {
```
Comment (SivanA-Kaltura) [inline src/kava.ts:893]: is this for IE ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (JonathanTGold) [reply package.json:37]: it is anyway not in use right now and not imported into src temporarily

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (JonathanTGold) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: i just formalizing the existing consensus

File: karma.conf.js (hunk 4 above)
Comment (JonathanTGold) [reply karma.conf.js:82]: restord

File: test/e2e/kava.spec.js (hunk 6 above)
Comment (JonathanTGold) [reply test/e2e/kava.spec.js:1]: fixed

File: src/kava.ts (hunk 8 above)
Comment (JonathanTGold) [reply src/kava.ts:893]: As a lesson from the previous refactor I do not change existing code

File: src/temp-imported-plugins-event-names-temp.ts (hunk 7 above)
Comment (JonathanTGold) [reply src/temp-imported-plugins-event-names-temp.ts:1]: Currently temporarily as a result of a refactor there is an incompatibility in the form of the import so currently plugins events has been moved to hard coded temporarily

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**